google-cloud-tools-plugin/testSrc/com/google/gct/idea/debugger/CloudDebugGlobalPollerTest.java

```diff
@@ -13,35 +19,77 @@
 import com.intellij.openapi.application.ApplicationManager;
 
 import org.hamcrest.beans.HasPropertyWithValue;
+import org.jetbrains.annotations.NotNull;
 import org.junit.Before;
 import org.junit.Test;
-import org.mockito.Mockito;
+
+import java.io.IOException;
 
 public class CloudDebugGlobalPollerTest extends BasePluginTestCase {
 
+  private static final String FAKE_USER_EMAIL = "foo@example.com";
+  private static final String FAKE_DEBUGGEE_ID = "debuggee-id-123456789";
+
   private CloudDebugProcessState cloudDebugProcessState;
+  private CloudDebugGlobalPoller cloudDebugGlobalPoller;
+  private Notifications notificationsHandler;
 
   @Before
   public void setUp() throws Exception {
     cloudDebugProcessState = new CloudDebugProcessState();
+    cloudDebugGlobalPoller = new CloudDebugGlobalPoller();
+    notificationsHandler = setupNotificationHandlerForVerification();
   }
 
   @Test
-  public void testPollForChangesFiresNotificationIfBackgroundListeningFails() throws Exception {
+  public void testPollForChangesFiresNotificationIfNoDebugClientObtained() {
     cloudDebugProcessState.setListenInBackground(true);
-    Application application = ApplicationManager.getApplication();
-    Notifications handler = Mockito.mock(Notifications.class);
-    // sending out notifications relies on several static method calls in
-    // com.intellij.notification.Notifications, let's subscribe to them and do verification this way
-    application.getMessageBus().connect(application).subscribe(Notifications.TOPIC,
-        handler);
 
-    CloudDebugGlobalPoller cloudDebugGlobalPoller = new CloudDebugGlobalPoller();
     cloudDebugGlobalPoller.pollForChanges(cloudDebugProcessState);
 
     assertFalse(cloudDebugProcessState.isListenInBackground());
-    verify(handler)
+    verifyNotificationFired();
+  }
+
+  @Test
+  public void testPollForChangesFiresNotificationOnIOException() throws IOException {
```

Comment:
the Google style naming convention for unit tests of a specific method are testMethodName_conditionInCamelCase(). I think in this case it would be something like testPollForChanges_firesNotificationIfNoDebugClientObtained()